### ISSUE TITLE
range_plus_one suggests using `RangeInclusive` where it is not valid

### ISSUE THREAD

**Issue by RDambrosio016** [rust-lang/rust-clippy#6561]
I tried this code:

```rust
use std::ops::Range;

trait Foo {};
// not implemented for RangeInclusive!
impl Foo for Range<usize> {}

fn foo(_: impl Foo) {}

fn main() {
  let idx = 1;
  // clippy suggests you use `idx..=idx` but that produces an error because RangeInclusive
  // is not accepted for `impl Foo`
  foo(idx..idx + 1);
}
```

I expected to see this happen: No warning

Instead, this happened: `an inclusive range would be more readable`

### Meta

- `cargo clippy -V`: clippy 0.0.212 (44e3daf5 2020-12-31)
- `rustc -Vv`:
```
rustc 1.51.0-nightly (44e3daf5e 2020-12-31)
binary: rustc
commit-hash: 44e3daf5eee8263dfc3a2509e78ddd1f6f783a0e
commit-date: 2020-12-31
host: x86_64-pc-windows-msvc
release: 1.51.0-nightly
```

**Comment by flip1995** [comment]
Duplicate of https://github.com/rust-lang/rust-clippy/issues/3307

**Comment by RDambrosio016** [comment]
Ah i apologize, i tried looking for a duplicate but couldnt find it 😄
